File: pkg/java/src/test/java/dev/openfga/language/util/ValidTransformerTestCase.java
package dev.openfga.language.util;

public final class ValidTransformerTestCase {
    private final String name;
    private final String dsl;
    private final String json;
    private final boolean skip;

    public ValidTransformerTestCase(String name, String dsl, String json, boolean skip) {
        this.name = name;
        this.dsl = dsl;
        this.json = json;
        this.skip = skip;
    }

    public String getName() {
        return name;
    }

    public String getDsl() {
        return dsl;
    }

    public String getJson() {
        return json;
    }

    public boolean isSkip() {
        return skip;
    }

    @Override
    public String toString() {
        return "TransformerTestCase[" + "name="
                + name + ", " + "dsl="
                + dsl + ", " + "json="
                + json + ", " + "skip="
                + skip + ']';
    }
}

File: pkg/csharp/test/OpenFga.Language.Tests/util/ValidTransformerTestCase.cs
using System.Text.Json.Serialization;

namespace OpenFga.Language.Tests.util;

public sealed class ValidTransformerTestCase
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("dsl")]
    public string Dsl { get; set; } = string.Empty;

    [JsonPropertyName("json")]
    public string Json { get; set; } = string.Empty;

    [JsonPropertyName("skip")]
    public bool Skip { get; set; }

    public ValidTransformerTestCase() { }

    public ValidTransformerTestCase(string name, string dsl, string json, bool skip)
    {
        Name = name;
        Dsl = dsl;
        Json = json;
        Skip = skip;
    }

    public override string ToString()
    {
        return $"TransformerTestCase[name={Name}, dsl={Dsl}, json={Json}, skip={Skip}]";
    }
}
